aptos-move/framework/aptos-framework/sources/PropertyList.move

```diff
@@ -0,0 +1,175 @@
+        let properties = PropertyList {
+           map: SimpleMap::create<String, PropertyValue>()
+        };
+        let i = 0;
+        while (i < Vector::length(default_keys)) {
+            SimpleMap::add(
+                &mut properties.map,
+                *Vector::borrow(default_keys, i),
+                PropertyValue {value: *Vector::borrow(default_values, i), type: *Vector::borrow(default_types, i)}
+            );
+            i  = i + 1;
+        };
+        properties
+    }
+    public fun contains(list: &PropertyList, key: &String): bool {
+        SimpleMap::contains_key(&list.map, key)
+    }
+
+    public fun add(list: &mut PropertyList, key: String, value: PropertyValue) {
+        assert!(! SimpleMap::contains_key(&list.map, &key), EKEY_AREADY_EXIST_IN_PROPERTY_MAP);
+        assert!(SimpleMap::length<String, PropertyValue>(&list.map) <= MAX_PROPERTY_MAP_SIZE, EPROPERTY_NUMBER_EXCEED_LIMIT);
+        SimpleMap::add(&mut list.map, key, value);
+
+    }
+
+    public fun size(list: &PropertyList): u64 {
+        SimpleMap::length(&list.map)
+    }
+
+    public fun borrow(list: &PropertyList, key: &String,): &PropertyValue {
+        let found = contains(list, key);
+        assert!(found, EPROPERTY_NOT_EXIST);
+        SimpleMap::borrow(&list.map, key)
+    }
+
+    public fun get_value(list: &PropertyList, key: &String,): vector<u8> {
+        let found = contains(list, key);
+        assert!(found, EPROPERTY_NOT_EXIST);
+        SimpleMap::borrow(&list.map, key).value
+    }
+
+    public fun get_type(list: &PropertyList, key: &String,): String {
+        let found = contains(list, key);
+        assert!(found, EPROPERTY_NOT_EXIST);
+        SimpleMap::borrow(&list.map, key).type
+    }
+
+    public fun get_property_values(
+        list: &PropertyList, keys: vector<String>): vector<PropertyValue> {
+        let res: vector<PropertyValue> = Vector::empty<PropertyValue>();
+        let i = 0;
+        while (i < Vector::length(&keys)) {
+            let key = Vector::borrow(&keys, i);
+            let value = borrow(list, key);
+            Vector::push_back(&mut res, *value);
+            i = i + 1;
+        };
+        res
+    }
+
+    public fun update_property_value(
+        list: &mut PropertyList,
+        key: &String,
+        value: PropertyValue ) {
+        let found = contains(list, key);
+        assert!(found, EPROPERTY_NOT_EXIST);
+        let property_val = SimpleMap::borrow_mut(&mut list.map, key);
+        *property_val = value;
+    }
+
+    public fun remove_property(
+        list: &mut PropertyList,
+        key: &String
+    ): (String, PropertyValue) {
+        let found = contains(list, key);
+        assert!(found, EPROPERTY_NOT_EXIST);
+
+        SimpleMap::remove(&mut list.map, key)
+    }
+
+    public fun update_property_values(
+        list: &mut PropertyList, keys: &vector<String>, values: vector<PropertyValue>){
```

Comment:
try to follow the more conventional formatting in Rust
```
  public fun a(
    param: ParamName,
    param: ParamName,
  ) {
```

Reply:
look at remove_property

Reply:
updated